src/Repository/Provider/InMemoryRepositoryProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Repository\Provider;
+
+use Repository\Repository;
+
+/**
+ * @author Ener-Getick <egetick@gmail.com>
+ */
+class InMemoryRepositoryProvider implements RepositoryProviderInterface
+{
+    private $repositories = array();
+
+    public function __construct(array $repositories)
+    {
+        foreach ($repositories as $repository) {
+            $this->addRepository($repository);
+        }
+    }
+
+    /**
+     * @param string $vendor
+     * @param string $name
+     *
+     * @return Repository|null
+     */
+    public function getRepository($vendor, $name)
+    {
+        if (isset($this->repositories[$fullName = strtolower($vendor.'/'.$name)])) {
+            return $this->repositories[$fullName];
+        }
+    }
+
+    /**
+     * Adds a repository.
+     *
+     * @param array $options
+     */
+    public function addRepository(array $options)
```

Comment:
It's to be coherent with the constrictor. In fact we could even use the Repository class here.